EIPS/eip-7783.md

```diff
@@ -0,0 +1,99 @@
+
+Add a new "Gas Target" selection strategy that takes in Block Number `N` and spits out the Gas Target `GT` for that block. The strategy is as follows:
+
+ - The gas target `GT_t` at block `t` is calculated as:
+
+```python
+def compute_gas_target(blockNum: int, blockNumStart: int, initialGasTarget: int, r: int, gasTargetCap: int) -> int:
+  if blockNum < blockNumStart:
+    return initialGasTarget
+  else:
+    return min(gasTargetCap, initialGasTarget + r * (blockNum - blockNumStart))
+```
+
+Where:
+
+- `blockNum` is the block number for which the gas target is being calculated.
+- `blockNumStart` is the block number at which the gas target increase starts.
+- `initialGasTarget` is the initial gas target at block `blockNumStart`.
+- `r` is the rate at which the gas target increases per block.
+- `gasTargetCap` is the maximum gas target that can be reached.
+
+If we set `blockNumStart` to the current block number, `initialGasTarget` to the current gas target (`15_000_000`), `r` to 3, and `gasTargetCap` to `30_000_000`, the gas target will increase by 3 gas per block for two years, reaching 30 million gas at the end of a 2 years period.
+
+### **Implementation**
+
+- **Client Updates:**
+  - Ethereum clients must implement the automatic gas target increase according to the specified rate and formula, with enforcement of the deactivation block. However, If the GasTarget has been locally set by the validator that will override the formula and take precedence.
+  
+- **Gas Target Management:**
+  - Miners and validators are not required to adjust the gas target manually during this period, as the increase will be automated at the protocol level.
+
+## **Rationale**
+
+### **Predictable Growth**
+
+- **Systematic Adjustment:**
+  - The gradual increase avoids sudden surges in gas target that could destabilize the network. Instead, it provides a smooth transition, giving the ecosystem time to adapt to larger block sizes.
+
+### **Controlled Target with Deactivation Block**
+
+- **Automatic Safeguard:**
+  - The inclusion of a deactivation block ensures that the gas target does not increase indefinitely, preventing potential negative impacts on network performance beyond the planned growth.
+
+
+- **Community Consensus:**
+  - The deactivation block serves only as a checkpoint for the community to evaluate the impact of the gas target increase, however, in the two-year period, the community can decide to halt the increase at any time and can also switch to a fixed gas target if needed.
+
+## **Backwards Compatibility**
+
+**No Hard Fork Required**
+
+## **Security Considerations**
+
+
+- The controlled gas target increase strategy is designed to prevent sudden changes that could lead to network instability or security vulnerabilities.
+- The fact that validators can re-adjust the gas target in case of a DDOS attack or other issues, makes the network more secure than to just increase the gas target manually in a cliff-like manner.
```

Comment:
```suggestion
- The fact that validators can re-adjust the gas target in case of a DOS attack or other issues, makes the network more secure than to just increasing the gas target manually in a cliff-like manner.
```